web/src/korrel8r/korrel8r.types.ts

```diff
@@ -9,7 +9,7 @@ export enum Korrel8rDomain {
 
 export abstract class Korrel8rNode {
   static fromURL: (url: string) => Korrel8rNode;
-  static fromQuery: (query: string) => Korrel8rNode;
+  static fromQuery: (query: string, start: number, end: number) => Korrel8rNode;
```

Comment:
```suggestion
  static fromQuery: (query: string, start: number | null, end: number | null) => Korrel8rNode;
```

If you change to the Constraint type instead just make sure to update the abstract class with the correct typing